static/app/views/dashboards/widgets/timeSeriesWidget/plottables/thresholds.tsx

```diff
@@ -0,0 +1,155 @@
+import type {SeriesOption} from 'echarts';
+import type {ItemStyleOption} from 'echarts/types/src/util/types';
+
+import MarkArea from 'sentry/components/charts/components/markArea';
+import MarkLine from 'sentry/components/charts/components/markLine';
+import type {Theme} from 'sentry/utils/theme';
+import type {
+  Thresholds as ThresholdsConfig,
+  TimeSeriesValueUnit,
+} from 'sentry/views/dashboards/widgets/common/types';
+import type {
+  Plottable,
+  PlottableTimeSeriesValueType,
+} from 'sentry/views/dashboards/widgets/timeSeriesWidget/plottables/plottable';
+
+type ThresholdPlottableOptions = {
+  thresholds: ThresholdsConfig;
+};
+
+type ThresholdPlottablePlottingOptions = {
+  theme: Theme;
+};
+
+export class Thresholds implements Plottable {
+  maxOffset = 5; // The offset from the top of the chart (in pixels), of the max threshold
+  thresholds: ThresholdsConfig;
+
+  /** State variables required for Plottable interface */
+  dataType: PlottableTimeSeriesValueType = 'duration';
+  dataUnit: TimeSeriesValueUnit = null;
+  isEmpty = false;
+  name = '';
+  needsColor = false;
+  start: number | null = null;
+  end: number | null = null;
+
+  constructor(options: ThresholdPlottableOptions) {
+    this.thresholds = options.thresholds;
+    this.isEmpty = !this.thresholds.max_values.max1 && !this.thresholds.max_values.max2;
```

Comment:
My only concern seeing this is that it seems like it's really bound to the implementation in Dashboards and not very generic. One day we want to be able to support a reverse order of shading (i.e. bad -> meh -> good). I think this is fine for now but it's a little bit of a constrained API for the generic components

I wonder if something more helpful is to pass in an array of thresholds and sort them, but I'm okay with that not being part of this PR and more of an improvement the next time we want to touch this stuff.